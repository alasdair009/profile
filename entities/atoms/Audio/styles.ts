"use client"
import styled from "styled-components";
import {colors} from "@/entities";

export const Root = styled.audio`
    appearance: none;
    
    &::-webkit-media-controls-panel {
        background: ${colors.greenGrass};
    }
    
    &::-webkit-media-controls-current-time-display,
    &::-webkit-media-controls-time-remaining-display {
        color: ${colors.whiteGhost};
    }
`;