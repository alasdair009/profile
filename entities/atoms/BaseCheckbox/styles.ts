"use client";

import styled from "styled-components";
import { colors, sizes } from "@/entities";
import { rem } from "polished";

export const Root = styled.input<{ $isInvalid: boolean }>`
  appearance: none;
  aspect-ratio: 1;
  background: ${colors.blackEvil};
  border-color: ${({ $isInvalid }) =>
    $isInvalid ? colors.redHeat : colors.greenGrass};
  border-style: solid;
  border-width: ${rem(1)};
  position: relative;
  width: ${sizes.s32.rem};

  &::after {
    background: ${colors.greenGrass};
    clip-path: polygon(0 0, 100% 0, 100% 50%, 37.5% 50%, 37.5% 100%, 0 100%);
    content: "";
    height: 50%;
    margin: 50%;
    position: absolute;
    transform: translateY(-50%) translateX(-100%) rotate(90deg) scale(0);
    transition: transform 300ms cubic-bezier(0.175, 0.885, 0.32, 1.275);
    width: 75%;
  }

  &:checked::after {
    transform: translateY(-60%) translateX(-50%) rotate(135deg) scaleX(-1);
  }

  &:user-invalid {
    border-color: ${colors.redHeat};
  }
`;
