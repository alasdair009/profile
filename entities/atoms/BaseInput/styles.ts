"use client";
import styled, { css } from "styled-components";
import { colors, fontSizes, globalInputMaxWidth, sizes } from "@/entities";
import { rem, rgba } from "polished";

const inValidStyles = {
  background: rgba(colors.redHeat, 0.15),
};

export const Root = styled.input<{
  $isInvalid: boolean;
  $hasBeenFocussed: boolean;
}>`
  appearance: none;
  background: ${({ $isInvalid }) =>
    $isInvalid ? inValidStyles.background : colors.greyDark};
  border: ${rem(1)} solid ${colors.greenGrass};
  color: ${colors.whiteGhost};
  display: block;
  font-size: ${fontSizes.medium.rem};
  margin: ${sizes.s2.rem} auto 0;
  max-width: ${rem(globalInputMaxWidth)};
  padding: ${sizes.s8.rem};
  width: 100%;

  &:user-invalid {
    background: ${({ $hasBeenFocussed }) =>
      $hasBeenFocussed ? inValidStyles.background : undefined};
  }
`;
