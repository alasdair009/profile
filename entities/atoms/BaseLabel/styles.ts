"use client";

import styled from "styled-components";
import { globalInputMaxWidth } from "../../design-tokens/dimensions";
import { rem } from "polished";

export const Root = styled.label`
  display: block;
  max-width: ${rem(globalInputMaxWidth)};
  text-align: center;
  width: 100%;
`;
