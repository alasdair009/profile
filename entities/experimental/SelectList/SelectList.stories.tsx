import { Meta, StoryObj } from "@storybook/react";
import { SelectList } from "./SelectList";
import { Figure } from "@/entities/experimental/SelectList/styles";
import { Button, HorizontalRule, Paragraph, sizes } from "@/entities";
import Image from "next/image";
import amLogo from "@/entities/assets/am.svg";
import { expect, userEvent, waitFor, within } from "@storybook/test";

const meta: Meta<typeof SelectList> = {
  component: SelectList,
  args: {
    defaultOption: "Select an option",
    children: (
      <>
        <option value="1">
          <Figure>
            <Image
              src={amLogo}
              alt="am Logo"
              height={sizes.s16.raw}
              width={sizes.s16.raw}
            />
          </Figure>
          <Paragraph>Option 1</Paragraph>
        </option>
        <option value="2">
          <Figure>
            <Image
              src={amLogo}
              alt="am Logo"
              height={sizes.s16.raw}
              width={sizes.s16.raw}
            />
          </Figure>
          <Paragraph>Option 2</Paragraph>
        </option>
        <option value="3">
          <Figure>
            <Image
              src={amLogo}
              alt="am Logo"
              height={sizes.s16.raw}
              width={sizes.s16.raw}
            />
          </Figure>
          <Paragraph>Option 3</Paragraph>
        </option>
        <HorizontalRule margin="none" />
        <option value="all">
          <Figure>
            <Image
              src={amLogo}
              alt="am Logo"
              height={sizes.s16.raw}
              width={sizes.s16.raw}
            />
          </Figure>
          <Paragraph>All Options</Paragraph>
        </option>
      </>
    ),
  },
  parameters: {
    docs: {
      description: {
        component:
          "An experimental new spec for a standardised styled dropdown box. \n\n _Please note to view this you must be viewing this story in Edge or Chrome with experimental features turned on._",
      },
    },
  },
};
export default meta;

export const Default: StoryObj<typeof SelectList> = {
  parameters: {
    chromatic: { disableSnapshot: true },
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const selectListElement = canvas.getByTestId(SelectList.name);
    const firstElement = canvas.getByText("Option 1")
      .parentElement as HTMLElement;

    await expect(selectListElement).toBeInTheDocument();
  },
};
