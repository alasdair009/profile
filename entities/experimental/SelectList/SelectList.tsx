import { HTMLAttributes } from "react";
import { Button, ListBox, MenuIcon, Option, Root, SelectedOption } from "./styles";
import { Paragraph } from "@/entities";

type SelectMenuProps = {
  /**
   *
   */
  defaultOption?: string;
} & HTMLAttributes<HTMLSelectElement>;
/**
 * An experimental new spec for a styled dropdown box.
 */
export function SelectList({ defaultOption, children, ...rest }: SelectMenuProps) {
  return (
    <Root as="selectlist" data-testid={SelectList.name} {...rest}>
      <Button type="selectlist">
        <SelectedOption as="selectedoption" />
      </Button>
      <ListBox as="listbox">
        <>
          {defaultOption && (
            <Option value="" hidden>
              <MenuIcon />
              <Paragraph>{defaultOption}</Paragraph>
            </Option>
          )}
          {children}
        </>
      </ListBox>
    </Root>
  );
}
