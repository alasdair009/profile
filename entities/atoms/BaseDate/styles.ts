"use client";
import styled from "styled-components";
import { defaultFieldStyles } from "@/entities/atoms/BaseInput/styles";
import { colors } from "@/entities";

export const Root = styled.input<{ $isInvalid: boolean }>`
  ${({ $isInvalid }) => defaultFieldStyles($isInvalid)}

  &::-webkit-calendar-picker-indicator {
    background-color: ${colors.greenGrass};
    cursor: pointer;

    &:hover {
      filter: brightness(1.3);
    }
  }
`;
