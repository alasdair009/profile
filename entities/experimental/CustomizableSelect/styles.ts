import styled from "styled-components";
import { defaultFieldStyles } from "@/entities/atoms/BaseInput/styles";
import { clipPaths, colors, sizes } from "@/entities";

export const Root = styled.select`
  ${defaultFieldStyles(false)}

  appearance: base-select;
  padding: 0;

  &::picker(select) {
    appearance: base-select;
  }

  option {
    align-items: center;
    background: ${colors.greyDark};
    color: ${colors.whiteGhost};
    display: flex;
    padding: ${sizes.s8.rem};
    width: 100%;

    &:hover {
      filter: brightness(1.3);
    }

    &::marker,
    &::before {
      display: none;
    }
  }
`;

export const SelectArea = styled.button`
  display: flex;
  justify-content: space-between;
  padding: 0;
  width: 100%;
`;

export const DropButton = styled.span`
  align-items: center;
  aspect-ratio: 1;
  background: ${colors.greenGrass};
  display: flex;
  height: 100%;
  justify-content: center;

  &:hover {
    filter: brightness(1.3);
  }

  &::before {
    aspect-ratio: 1;
    background: ${colors.whiteGhost};
    clip-path: ${clipPaths.chevronDown};
    content: "";
    height: ${sizes.s32.rem};
  }
`;

export const SelectedContent = styled.div`
  align-items: center;
  display: flex;
  width: 100%;
`;
