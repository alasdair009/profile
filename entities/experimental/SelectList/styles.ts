"use client";
import styled, { css } from "styled-components";
import { borderRadii, colors, sizes } from "@/entities";
import { darken, lighten, rgba } from "polished";
import { fieldHeight } from "@/entities/atoms/BaseInput/styles";

const optionStyles = css`
  align-items: center;
  box-sizing: border-box;
  color: ${colors.whiteGhost};
  cursor: pointer;
  display: flex;
  gap: ${sizes.s8.rem};
  justify-content: center;
  transition-duration: 0.25s;
  width: 100%;

  * {
    margin-bottom: 0;
    margin-top: 0;
  }
`;

export const Root = styled.select`
  min-height: ${fieldHeight};
  max-width: ${sizes.s256.rem};
  position: relative;
  width: 100%;

  &:hover {
    filter: brightness(1.3);
  }

  &:has(:popover-open) button {
    background: ${darken(0.1, colors.greenGrass)};
  }

  option {
    ${optionStyles};
    background-color: ${colors.blackEvil};
    padding: ${sizes.s8.rem};

    &[hidden] {
      display: none;
    }

    &:hover {
      background-color: ${lighten(0.1, colors.blackEvil)};
    }
  }
`;

export const Button = styled.button<{ type: "selectlist" }>`
  background: ${colors.greenGrass};
  color: ${colors.whiteGhost};
  cursor: pointer;
  transition-duration: 0.25s;

  width: 100%;
`;

export const SelectedOption = styled.span`
  ${optionStyles};

  figure {
    background: ${rgba(colors.blackEvil, 0.9)};
  }
`;

export const ListBox = styled.div`
  padding: 0;
`;

export const MenuIcon = styled.span`
  aspect-ratio: 1;
  background: ${colors.whiteGhost};
  clip-path: polygon(0 0, 100% 0, 100% 20%, 0 20%, 0 40%, 100% 40%, 100% 60%, 0 60%, 0 80%, 100% 80%, 100% 100%, 0 100%);
  display: flex;
  width: ${sizes.s24.rem};
`;

export const Option = styled.option`
  ${optionStyles};
  background-color: ${colors.blackEvil};
  padding: ${sizes.s8.rem};

  &[hidden] {
    display: none;
  }

  &:hover {
    background-color: ${lighten(0.1, colors.blackEvil)};
  }
`;

export const Figure = styled.figure`
  align-items: center;
  aspect-ratio: 1;
  border-radius: ${borderRadii.round};
  display: flex;
  justify-content: center;
  margin: 0;
  padding: ${sizes.s2.rem};
  overflow: hidden;
  position: relative;
  width: ${sizes.s24.rem};
`;
