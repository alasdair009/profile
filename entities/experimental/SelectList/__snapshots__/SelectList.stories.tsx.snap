// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`experimental/SelectList Default play-test 1`] = `
<selectlist data-testid="SelectList"
            class="sc-kGqRax iIDRRV"
>
  <button type="selectlist"
          class="sc-dCKixA hrsSZ"
  >
    <selectedoption class="sc-giOWAb gXVkfv">
    </selectedoption>
  </button>
  <listbox class="sc-kuDXwa hwzoru">
    <option value
            hidden
            class="sc-catHVh hKOVEB"
    >
      <span class="sc-llIhno kmzfQL">
      </span>
      <p data-testid="Paragraph"
         class="sc-gLXSEc fjewFw"
      >
        Select an option
      </p>
    </option>
    <option value="1">
      <figure class="sc-eiLgtK cxDvHx">
        <img alt="am Logo"
             loading="lazy"
             width="16"
             height="16"
             decoding="async"
             data-nimg="1"
             srcset="/static/media/entities/assets/am.svg?w=16&amp;q=75 1x, /static/media/entities/assets/am.svg?w=32&amp;q=75 2x"
             src="/static/media/entities/assets/am.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
      </figure>
      <p data-testid="Paragraph"
         class="sc-gLXSEc fjewFw"
      >
        Option 1
      </p>
    </option>
    <option value="2">
      <figure class="sc-eiLgtK cxDvHx">
        <img alt="am Logo"
             loading="lazy"
             width="16"
             height="16"
             decoding="async"
             data-nimg="1"
             srcset="/static/media/entities/assets/am.svg?w=16&amp;q=75 1x, /static/media/entities/assets/am.svg?w=32&amp;q=75 2x"
             src="/static/media/entities/assets/am.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
      </figure>
      <p data-testid="Paragraph"
         class="sc-gLXSEc fjewFw"
      >
        Option 2
      </p>
    </option>
    <option value="3">
      <figure class="sc-eiLgtK cxDvHx">
        <img alt="am Logo"
             loading="lazy"
             width="16"
             height="16"
             decoding="async"
             data-nimg="1"
             srcset="/static/media/entities/assets/am.svg?w=16&amp;q=75 1x, /static/media/entities/assets/am.svg?w=32&amp;q=75 2x"
             src="/static/media/entities/assets/am.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
      </figure>
      <p data-testid="Paragraph"
         class="sc-gLXSEc fjewFw"
      >
        Option 3
      </p>
    </option>
    <hr data-testid="HorizontalRule"
        class="sc-ifyrTC jazMDl"
    >
    <option value="all">
      <figure class="sc-eiLgtK cxDvHx">
        <img alt="am Logo"
             loading="lazy"
             width="16"
             height="16"
             decoding="async"
             data-nimg="1"
             srcset="/static/media/entities/assets/am.svg?w=16&amp;q=75 1x, /static/media/entities/assets/am.svg?w=32&amp;q=75 2x"
             src="/static/media/entities/assets/am.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
      </figure>
      <p data-testid="Paragraph"
         class="sc-gLXSEc fjewFw"
      >
        All Options
      </p>
    </option>
  </listbox>
</selectlist>
`;
